src/shared/components/query/GenesetsVolcanoSelector.tsx

```diff
@@ -0,0 +1,313 @@
+            },
+            'pValue': {
+                priority: 3,
+                dataField: 'representativePvalue',
+                name: "P Value",
+                sortable: true,
+                filterable: true,
+                formatter: ({name, columnData: value}: IColumnFormatterData<Geneset>) => (
+                    <Td key={name} column={name} value={value}>
+                    {toPrecision(value, 2, 0.1)}
+                    </Td>
+                ),
+            },
+            'selected': {
+                name: "Selected",
+                priority: 4,
+                sortable: false,
+                filterable: false,
+                formatter: ({name, rowData: geneset}:IColumnFormatterData<Geneset>) => (
+                    <Td key={name} column={name}>
+                    {!!(geneset) && (
+                            <div className={styles.selectionColumnCell}>
+                            <Observer>
+                            {() => (
+                                <LabeledCheckbox
+                                checked={!!this.map_genesets_selected.get(geneset.name)}
+                                onChange={event => (this.map_genesets_selected.set(geneset.name, event.target.checked)) }
+                            />
+                            )}
+                    </Observer>
+                    </div>
+                    )}
+                    </Td>
+                ),
+            },
+    };
+    
+    async fetchData() {
+        const tableData: Geneset[] = [];
+        const plotData: {x: number, y: number, fill: string}[] = [];
+        const hierarchyData = await getHierarchyData(
+                this.props.gsvaProfile, Number(this.selectedPercentile.value),0, 1, this.props.sampleListId);
+        for (const node of hierarchyData) {
+            if (_.has(node, 'genesets')) {
+                for (const geneset of node.genesets) {
+                    tableData.push({
+                        genesetId: geneset.genesetId,
+                        name: geneset.genesetId,
+                        description : geneset.description,
+                        representativeScore : geneset.representativeScore,
+                        representativePvalue : geneset.representativePvalue,
+                        refLink : geneset.refLink,
+                    });
+                }
+            }
+        }
+        
+        this.tableData = tableData;
+        this.allTableData = tableData;
+        this.isLoading = false;
+    }
+    
+    percentileChange(val: {label: string, value: string} | null)
+    {
+        this.selectedPercentile = val || {label: '75%', value: '75'};
+        this.percentileHasChanged = true;
+    }
+    
+    updateSelectionFromPlot(points: any, bounds: any) {
+        const selectedPoints = [];
+        const newTableData: Geneset[] = [];
+        const newPlotData = this.state.plotData;
+        this.map_genesets_selected = new ObservableMap<boolean>();
```

Comment:
You probably don't need an ObservableMap, which is more costly.  For example, it's going to issue synchronous update events every time you call set down below.  Instead, build the table as a plain object and then either assign it to a observable.ref (look that up) or THEN instantiate an Observable Map with it if you really need to functionality of an ObservableMap